# HG changeset patch
# User Jacob Dawid <jacob.dawid@googlemail.com>
# Date 1303920608 -7200
#      Wed Apr 27 18:10:08 2011 +0200
# Node ID 90d4d110721b5188bf194326b7c5e197e7fd64ee
# Parent  cb80550ebaed96bf058880304ee27bdfb803e629
Single qouted strings are recognized.

diff --git a/gui//src/OctaveLink.cpp b/gui//src/OctaveLink.cpp
--- a/gui//src/OctaveLink.cpp
+++ b/gui//src/OctaveLink.cpp
@@ -37,18 +37,22 @@ OctaveLink::~OctaveLink() {
 }
 
 int OctaveLink::readlineEventHook() {
   OctaveLink::instance()->processOctaveServerData();
   return 0;
 }
 
 QString OctaveLink::octaveValueAsQString(OctaveValue octaveValue) {
-    // Convert string.
-    if(octaveValue.is_string()) {
+    // Convert single qouted string.
+    if(octaveValue.is_sq_string()) {
+        return QString("\'%1\'").arg(octaveValue.string_value().c_str());
+
+    // Convert double qouted string.
+    } else if(octaveValue.is_dq_string()) {
         return QString("\"%1\"").arg(octaveValue.string_value().c_str());
 
     // Convert real scalar.
     } else if(octaveValue.is_real_scalar()) {
         return QString("%1").arg(octaveValue.scalar_value());
 
     // Convert complex scalar.
     } else if(octaveValue.is_complex_scalar()) {
